lib/active_record_shards/default_replica_patches.rb

```diff
@@ -0,0 +1,171 @@
+# frozen_string_literal: true
+
+module ActiveRecordShards
+  module DefaultReplicaPatches
+    def self.wrap_method_in_on_replica(class_method, base, method, force_on_replica: false)
+      base_methods =
+        if class_method
+          base.methods + base.private_methods
+        else
+          base.instance_methods + base.private_instance_methods
+        end
+
+      return unless base_methods.include?(method)
+
+      _, method, punctuation = method.to_s.match(/^(.*?)([\?\!]?)$/).to_a
+      # _ALWAYS_ on replica, or only for on `on_replica_by_default = true` models?
+      wrapper = force_on_replica ? 'force_on_replica' : 'on_replica_unless_tx'
+      base.class_eval <<-RUBY, __FILE__, __LINE__ + 1
+        #{class_method ? 'class << self' : ''}
+          def #{method}_with_default_replica#{punctuation}(*args, &block)
+            #{wrapper} do
+              #{method}_without_default_replica#{punctuation}(*args, &block)
+            end
+          end
+
+          alias_method :#{method}_without_default_replica#{punctuation}, :#{method}#{punctuation}
+          alias_method :#{method}#{punctuation}, :#{method}_with_default_replica#{punctuation}
+        #{class_method ? 'end' : ''}
+      RUBY
+    end
+
+    def self.wrap_method_in_on_slave(*args)
+      ActiveRecordShards::Deprecation.deprecation_warning(
+        :'self.wrap_method_in_on_slave',
+        :'self.wrap_method_in_on_replica'
+      )
+      wrap_method_in_on_replica(*args)
+    end
+
+    def columns_with_force_replica(*args, &block)
+      on_cx_switch_block(:replica, construct_ro_scope: false, force: true) do
+        columns_without_force_replica(*args, &block)
+      end
+    end
+    alias_method :columns_with_force_slave, :columns_with_force_replica
+
+    def table_exists_with_force_replica?(*args, &block)
+      on_cx_switch_block(:replica, construct_ro_scope: false, force: true) do
+        table_exists_without_force_replica?(*args, &block)
+      end
+    end
+    alias_method :table_exists_with_force_slave?, :table_exists_with_force_replica?
+
```

Comment:
These two methods were removed in 790fb4754ebba5370fe60cc52f04d57b00bf5f37, but were accidentally added back in this PR.

This mistake will be remedied in #254, if not sooner.